tests/integration/components/signup-steps/step-one-test.js

```diff
@@ -0,0 +1,79 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'website-www/tests/helpers';
+import { render, typeIn, fillIn } from '@ember/test-helpers';
+import { set } from '@ember/object';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | signup-steps/step-one', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('render firstname input field', async function (assert) {
+    assert.expect(14);
+    this.set('name', 'firstname');
+    this.set('field', 'First Name');
+    this.set('placeHolder', 'Write your first name');
+    this.set('type', 'text');
+    this.set('required', true);
+    this.set('value', '');
+    this.set('disabled', false);
+    this.set('onInput', (e) => {
+      // Update the input value when the input changes
+      this.value = e.target.value;
+      // Call handleInputChange with the current value
+      this.handleInputChange('firstname', this.value);
+    });
+
+    this.set('handleInputChange', (actual1, actual2) => {
+      this.name = actual1;
+      this.value = actual2;
+    });
+
+    await render(
+      hbs`<SignupSteps::StepOne @onChange={{this.handleInputChange}}/>`
+    );
+
+    assert.dom('[data-test-input]').hasClass('input-box');
+
+    assert.dom('[data-test-label]').hasClass('label');
+    assert.dom('[data-test-label]').hasText('First Name');
+    assert.dom('[data-test-label]').hasAttribute('for', 'firstname');
+
+    assert.dom('[data-test-required]').hasClass('required');
+
+    assert.dom('[data-test-input-field]').hasClass('input__field');
+    assert.dom('[data-test-input-field]').hasAttribute('required');
+    assert.dom('[data-test-input-field]').hasAttribute('name', 'firstname');
+    assert.dom('[data-test-input-field]').hasProperty('type', 'text');
+    assert.dom('[data-test-input-field]').hasAttribute('id', 'firstname');
+    assert
+      .dom('[data-test-input-field]')
+      .hasProperty('placeholder', 'Write your first name');
+    assert.dom('[data-test-input-field]').hasProperty('value', '');
+    await typeIn('[data-test-input-field]', 'shubham');
+    assert.strictEqual(this.name, 'firstname');
+    assert.strictEqual(this.value, 'shubham');
+  });
+
+  test('it updates firstname in signupDetails when handleInputChange is triggered', async function (assert) {
+    assert.expect(1);
+    this.set('signupDetails', {
+      firstname: '',
+    });
+
+    this.set('handleInputChange', (key, value) => {
+      set(this.signupDetails, key, value);
+    });
+
+    await render(hbs`
+    <SignupSteps::StepOne @onChange={{this.handleInputChange}} />
+  `);
+
+    await fillIn('input', 'shubham');
```

Comment:
Can we have data-test selector here?

Reply:
I used the term "fillIn" because I thought it was filling in a property value of the signupObject. However, upon reflection, I believe that "typeIn" might be a more appropriate term because it represents the action of passing a value to a property in the signupObject as I am typing.